### PR TITLE
[feat] add Ubuntu 24.04 Noble package support

### PR DESCRIPTION
**Pull request by flyinprogrammer**
In theory this _might_ capture this [request](https://github.com/valkey-io/valkey/pull/315#discussion_r1575245988) for an issue! 🌮

### CODE REVIEW

**Review by roshkhatri — APPROVED** [review]
LGTM

**Review by melroy89 — APPROVED** [review]
Not sure why the CICD test fails. But this change LGTM. Thx!

**Comment by madolson** [comment]
> Not sure why the CICD test fails. But this change LGTM. Thx!

We had some instability around when we launched Valkey 8 because of the features. You can find most of the documented issues here: https://github.com/valkey-io/valkey/issues?q=is%3Aopen+is%3Aissue+label%3Atest-failure. The CI pretty stable generally.

File: utils/releasetools/build-config.json [hunk 1]
```diff
@@ -12,6 +12,12 @@
           "type": "deb",
           "platform": "focal"
         },
+        {
+          "arch": "x86_64",
+          "target": "ubuntu24.04",
```
Comment (madolson) [inline utils/releasetools/build-config.json:17]: @roshkhatri I see we are skipping 22.04.  I'm Okay adding these, but maybe we should document our strategy here somewhere?

**Review by madolson — APPROVED** [review]

File: utils/releasetools/build-config.json (hunk 1 above)
Comment (melroy89) [reply utils/releasetools/build-config.json:17]: Like focus on Ubuntu LTS releases would make sense to me.

Comment (madolson) [reply utils/releasetools/build-config.json:17]: @roshkhatri @bbarani Can we make an issue to document our strategy and follow through on updating the website. I think I'll merge this now and we can follow up on the next release adding these images.

Comment (roshkhatri) [reply utils/releasetools/build-config.json:17]: Yes I have opened an issue for the website repo here https://github.com/valkey-io/valkey-io.github.io/issues/164.

Comment (roshkhatri) [reply utils/releasetools/build-config.json:17]: Also I think there is a problem, for compiling, we use https://github.com/marketplace/actions/run-on-architecture for the binaries for ARM architecture. Github ARM runners are in GA but as github-hosted runners, there are not in opensource according to https://github.com/orgs/community/discussions/19197#discussioncomment-10550689.

I  will look into more options.

**Pull request merged**